core/local/change.js

```diff
@@ -274,33 +274,26 @@ export type LocalMove = LocalFileMove|LocalDirMove
 export type LocalMoveEvent = LocalFileAdded|LocalDirAdded
 */
 
-function InvalidLocalMoveEvent (moveChange /*: LocalMove */, event /*: LocalMoveEvent */) {
-  this.name = 'InvalidLocalMoveEvent'
-  this.moveChange = moveChange
-  this.event = event
-  // FIXME: Include event/change details in message
-  this.message = `Cannot include event ${event.type} into change ${moveChange.type}`
-  Error.captureStackTrace(this, this.constructor)
-}
-
-const ensureValidMoveEvent = (moveChange /*: LocalMove */, event /*: LocalMoveEvent */) => {
-  /* istanbul ignore next */
-  if (!moveChange.wip) throw new InvalidLocalMoveEvent(moveChange, event)
-}
-
 function includeAddEventInFileMove (moveChange /*: LocalFileMove */, e /*: LocalFileAdded */) {
   if (!moveChange.wip &&
        moveChange.path === e.path &&
        moveChange.stats.ino === e.stats.ino &&
        moveChange.md5sum === e.md5sum) return
-  ensureValidMoveEvent(moveChange, e)
   moveChange.path = e.path
   moveChange.stats = e.stats
   moveChange.md5sum = e.md5sum
-  delete moveChange.wip
-  log.debug(
-    {path: e.path, oldpath: moveChange.old.path, ino: moveChange.stats.ino},
-    'FileMove + add = FileMove')
+
+  if (e.md5sum) {
```

Comment:
Maybe we could later use the exact same `!e.wip` condition as in `includeAddDirEventInDirMove()` below.
The logic looks the same (possibly easier to read by reversing the if/else branches in both cases).
Not a blocker anyway :slightly_smiling_face:

Reply:
Yes I thought about the condition itself but I wanted the previous behavior to remain in the first branch. I'm not sure it's useful though :).

As for using `e.wip`, it would be nice but I wasn't sure it was always set at the time.